config/crd/components/all/more_validations_in_bindingpolicies.yaml

```diff
@@ -0,0 +1,12 @@
+- op: add
+  path: /spec/versions/0/schema/openAPIV3Schema/properties/spec/properties/clusterSelectors/items/properties/matchLabels/additionalProperties/pattern
+  value: "^(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?$"
```

Comment:
This does not express the length restriction.

Reply:
I don’t see a way to accurately check the length of the value using regex.
 
Also, this regex here is the same one used by the apiserver to validate label values, and the apiserver checks the length before and in addition to using the regex: https://github.com/kubernetes/kubernetes/blob/ebc1a7b7fb52f30021b7e66f77070dac8e6b839b/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L166-L176

Reply:
O crud, I see that Go regex does not have an "AND" constructor. So indeed you can not write a regex that checks length AND syntax.